lib/vorbisfile.c

```diff
@@ -1989,7 +1989,8 @@ long ov_read_filter(OggVorbis_File *vf,char *buffer,int length,
     long channels=ov_info(vf,-1)->channels;
     long bytespersample=word * channels;
     vorbis_fpu_control fpu;
-    if(samples>length/bytespersample)samples=length/bytespersample;
+    if(bytespersample && samples>length/bytespersample)
+      samples=length/bytespersample;
```

Comment:
I think it would be better to return `OV_EINVAL` here. Neither `word` nor `channels` should be zero here.